src/Settings/index.js

```diff
@@ -13,4 +18,15 @@ export default {
   has(type, key) {
     return settings[type] && key in settings[type]
   },
+  set(key, value, type = 'user') {
+    settings[type] = settings[type] || {}
+    settings[type][key] = value
+    publish(type, key, value)
+  },
+  subscribe(callback) {
```

Comment:
I think it would be nicer if we could pass a `key` to the `subscribe` method, so the callback only gets invoked whenever that specific setting changes.

Reply:
Pondering it - Redux follows the subscribe to store changes and you figure out what changed philosophy. Maybe we can flatten the Settings to be only two levels deep and drop the "type", otherwise this becomes `subscribe(type, key, callback)` which is strange since app / platform can't change.

Reply:
I think we should omit the possibility to pass a type all together. Although I'd like to keep storing user defined settings under the user-type. So that basically becomes a hardcoded type.

And since only user defined settings ever call the publish method, I don't think there's a need to explicitly pass a type here either.

My suggestion would actually be to store the callbacks with the key they subscribe to, so we can prevent from looping through all registered callback when any key changes.

Reply:
How would you change the getter?

```
get(type, key) {
    return settings[type] && settings[type][key]
  },
  has(type, key) {
    return settings[type] && key in settings[type]
  },
```

Would someone have to do `get('user', 'key')` - how would they know the type user? would that be part of the docs?

Reply:
I would keep the `get`-method _as is_ and have developers pass 'user' as a `type`. That would be necessary anyway, even in the case where you'd set a setting without explicitly passing a type.

And yes, agreed this would have to be properly documented. But it does make sense that you'd 'get' user-defined settings by passing 'user' a type.